src/StoreApi/Routes/AbstractCartRoute.php

```diff
@@ -36,7 +36,9 @@ protected function maybe_recalculate_totals() {
 				]
 			)
 		);
-		if ( ! hash_equals( $current_hash, $new_hash ) ) {
+		$changed      = ! hash_equals( $current_hash, $new_hash );
+
+		if ( apply_filters( '__experimental_woocommerce_store_api_cart_recalculate_totals', $changed ) ) {
```

Comment:
Subs requires a calculation otherwise the recurring cart data is all missing. Looks like we may not need this if we remove the cache - other things like fees also cannot be cached because they are dynamic https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3678